clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubMessageHelper.java

```diff
@@ -70,12 +73,78 @@ private static GoogleCredentials getCredentials() {
     /**
      * Gets the topic name with the given prefix.
      *
-     * @param topic The endpoint of the topic (e.g. event)
-     * @param prefix The prefix of the Pub/Sub topic, it's either the tenant ID or the adapter instance ID
+     * @param topic The endpoint of the topic (e.g. event).
+     * @param prefix The prefix of the Pub/Sub topic, it's either the tenant ID or the adapter instance ID.
      * @return The topic containing the prefix identifier and the endpoint.
      */
     public static String getTopicName(final String topic, final String prefix) {
-        return String.format("%s.%s", prefix, topic);
+        return getTopicName(topic, prefix, new ArrayList<>());
+    }
+
+    /**
+     * Gets the topic name with the given prefix and subtopics.
+     *
+     * @param topic The endpoint of the topic (e.g. event).
+     * @param prefix The prefix of the Pub/Sub topic, it's either the tenant ID or the adapter instance ID.
+     * @param subtopics A list with the subtopics.
+     * @return The topic containing the prefix identifier, the endpoint and all the subtopics.
+     */
+    public static String getTopicName(final String topic, final String prefix, final List<String> subtopics) {
+        final StringBuilder topicBuilder = new StringBuilder();
+        topicBuilder.append(prefix).append(".").append(topic);
+        for (String subtopic : subtopics) {
+            topicBuilder.append(".").append(subtopic);
+        }
+        return topicBuilder.toString();
+    }
+
+    /**
+     * Gets the subtopics from the message attributes.
+     *
+     * @param attributesMap The attribute map.
+     * @return An ordered list containing all the subtopics.
```

Comment:
... or an empty list if the topic has no subtopics.

Reply:
every list is ordered, the question is: what is it ordered by?

Reply:
good point, I updated the description.